sources/zafira-web/src/main/java/com/qaprosoft/zafira/web/DashboardController.java

```diff
@@ -85,6 +90,9 @@ public DashboardType createDashboard(@RequestBody @Valid DashboardType dashboard
     @Override
     public DashboardType getDashboardById(@PathVariable("id") long id) {
         Dashboard dashboard = dashboardService.getDashboardById(id);
+        if (dashboard.isHidden() && !hasPermission(Permission.Name.VIEW_HIDDEN_DASHBOARDS)) {
```

Comment:
In which case is this condition true assuming PreAuthorize conditions works correctly?